modules/account-lib/src/coin/eth/types.ts

```diff
@@ -25,22 +35,41 @@ export class EthTransactionData implements EthLikeTransactionData {
    * @returns {EthTransactionData} a new ethereum transaction object
    */
   public static fromJson(tx: TxData, common: EthereumCommon): EthTransactionData {
+    const nonce = addHexPrefix(new BigNumber(tx.nonce).toString(16));
+    const value = addHexPrefix(new BigNumber(tx.value).toString(16));
+    const gasLimit = addHexPrefix(new BigNumber(tx.gasLimit).toString(16));
+    const chainId = tx.chainId ? addHexPrefix(new BigNumber(tx.chainId).toString(16)) : undefined;
+
+    const gasPrice = isLegacyTx(tx) ? addHexPrefix(new BigNumber(tx.gasPrice).toString(16)) : undefined;
+
+    const maxFeePerGas = isEIP1559Txn(tx) ? addHexPrefix(new BigNumber(tx.maxFeePerGas).toString(16)) : undefined;
+    const maxPriorityFeePerGas = isEIP1559Txn(tx)
+      ? addHexPrefix(new BigNumber(tx.maxPriorityFeePerGas).toString(16))
+      : undefined;
+
     return new EthTransactionData(
-      new EthereumTx(
+      TransactionFactory.fromTxData(
         {
-          nonce: addHexPrefix(new BigNumber(tx.nonce).toString(16)),
+          type: isLegacyTx(tx) ? LEGACY_TX_TYPE : EIP1559_TX_TYPE,
```

Comment:
will we always have either legacy or eip1559? want to make sure we dont allow type 1 to be misclassified if other methods won't first error with tx_type 1 for example

Reply:
Yes because TxData is a type we defined where the type is either legacy or eip1559. it would only be an issue when deserializing a tx. which i handled by throwing an error